hat/test/org/hatdex/hat/she/service/FunctionServiceContext.scala

```diff
@@ -43,17 +42,16 @@ trait FunctionServiceContext extends HATTestContext {
   /**
     * A fake Guice module.
     */
-  class CustomisedFakeModule extends AbstractModule with ScalaModule {
+  class CustomisedFakeModule extends ScalaModule {
 
     @Provides
     def provideFunctionExecutableRegistry(): FunctionExecutableRegistry =
       new FunctionExecutableRegistry(Seq(registeredFunction, registeredDummyFunction, registeredDummyFunctionAvailable))
   }
 
   override lazy val application: Application = new GuiceApplicationBuilder()
-    .configure(FakeHatConfiguration.config)
-    .overrides(new FakeModule)
-    .overrides(new ExtrasModule)
+    .configure(conf)
```

Comment:
Curious question again, is `conf` some kind of global variable? Like it automatically passes in the configuration file that used when running the application?

Reply:
it comes from the parent class of most HAT integration tests namely `HATTestContext`
it only contains the dynamic parts of the config i.e. the ports of the docker containers running for a given tests (each test will have different ones)